### ISSUE TITLE
Corp.js eventually stops upgrading warehouses

### ISSUE THREAD

**Issue by cjvalcho** [kamukrass/Bitburner#4]
Despite Healthcare having a higher potential profit than Tobacco, I've noticed that Tobacco eventually produces far more income than Healthcare using this script. ~~This is due to Healthcare's production being throttled by a lack of warehouse space to store all of the produced goods to sell.~~ Healthcare may just scale poorly compared to Tobacco. It _does_ have issues with warehouses, though.

I lowered the threshold to upgrade a warehouse from 90% to 75% as a test. While this did result in the script purchasing new warehouses, after a short time the script eventually stopped purchasing warehouses once more. In my current run, Healthcare's warehouses are 100% filled, but the script only rarely purchases upgrades.

**Comment by kamukrass** [comment]
One improvement potential especially when you have bonus time is that hiring employees and setting them to new jobs takes a lot of time (several seconds). Other upgrades thus only increment by 1 every few seconds and especially for warehouses their storage might be queried just while they are empty after selling (and thus not increased). So warehouse space might grow slower than it should in some situations; but it grows. 
In your scenario - does the warehouse space issue make a significant impact, like do you really need more money (and waiting a few minutes does not provide it)?

I assume the profitability of industries has a random factor in. I have seen Tobacco and Healthcare ahead sometimes.

**Comment by cjvalcho** [comment]
It's unlikely that there's any real need for more money, as the Tobacco industry is making many times the income of the Healthcare industry. It's more a case of the game running slowly and the script responding slowly, I suppose. This took place outside of bonus time, where the Corporation had a massive amount of money, and warehouses were still below a single trillion to improve, but it'd take upwards of 45 seconds for it to apparently realize that it needed to upgrade the warehouse.

I've only used this script a few times since I've started making corporations, and in each time Tobacco pulls ahead massively from Healthcare. I'm not entirely sure why that is, though. Bad luck, perhaps?
